src/transformers/models/iris/modeling_iris.py

```diff
@@ -0,0 +1,1964 @@
+                    )
+                )
+                block_in = block_out
+                if curr_res in config.attn_resolutions:
+                    attn.append(AttnBlock(block_in))
+            up = nn.Module()
+            up.block = block
+            up.attn = attn
+            if i_level != 0:
+                up.upsample = Upsample(block_in, with_conv=True)
+                curr_res = curr_res * 2
+            self.up.insert(0, up)  # prepend to get consistent order
+
+        # end
+        self.norm_out = Normalize(block_in)
+        self.conv_out = torch.nn.Conv2d(block_in, config.out_ch, kernel_size=3, stride=1, padding=1)
+
+    def forward(
+        self, z: torch.Tensor, output_hidden_states: bool = False, output_attentions: bool = False
+    ) -> tuple[torch.Tensor, list[torch.Tensor]]:
+        temb = None  # timestep embedding
+
+        # z to block_in
+        h = self.conv_in(z)
+        hidden_states = () if output_hidden_states else None
+        attentions = () if output_attentions else None
+
+        # middle
+        h = self.mid.block_1(h, temb)
+        # Add the hidden state to the tuple
+        hidden_states = hidden_states + (h,) if output_hidden_states else None
+        attn_out = self.mid.attn_1(h)
+        h = attn_out[0]
+        # Add attention weights
+        attentions = attentions + (attn_out[1],) if output_attentions else None
+
+        # Add the hidden state to the tuple
+        hidden_states = hidden_states + (h,) if output_hidden_states else None
+        h = self.mid.block_2(h, temb)
+        # Add the hidden state to the tuple
+        hidden_states = hidden_states + (h,) if output_hidden_states else None
+
+        # upsampling
+        for i_level in reversed(range(self.num_resolutions)):
+            for i_block in range(self.config.num_res_blocks + 1):
+                h = self.up[i_level].block[i_block](h, temb)
+                # Add the hidden state to the tuple
+                hidden_states = hidden_states + (h,) if output_hidden_states else None
+                if len(self.up[i_level].attn) > 0:
+                    attn_out = self.up[i_level].attn[i_block](h)
+                    h = attn_out[0]
+                    # Add attention weights
+                    attentions = attentions + (attn_out[1],) if output_attentions else None
+                    # Add the hidden state to the tuple
+                    hidden_states = hidden_states + (h,) if output_hidden_states else None
+            if i_level != 0:
+                h = self.up[i_level].upsample(h)
+                # Add the hidden state to the tuple
+                hidden_states = hidden_states + (h,) if output_hidden_states else None
+
+        # end
+        h = self.norm_out(h)
+        h = nonlinearity(h)
+        h = self.conv_out(h)
+
+        return h, hidden_states, attentions
+
+
+def nonlinearity(x: torch.Tensor) -> torch.Tensor:
+    # swish
```

Comment:
You can use the swish activation already defined in the library

Reply:
oh!sure, thanks for mentioning :+1: